script/misc-helper.ts

```diff
@@ -375,20 +375,29 @@ async function getCIHelper(): Promise<CIHelper> {
             await set({ appID: 46807, name: org});
         }
     } else if (command === "handle-pr-comment") {
-        if (commander.args.length !== 2) {
-            process.stderr.write(`${command}: Expected one comment ID\n`);
+        if (commander.args.length !== 2 && commander.args.length !== 3) {
+            process.stderr.write(`${command}: optionally takes  a ${
```

Comment:
`handle-pr-push` says `need` and not `optionally`.  I assume the gitgitgadget pipeline will be updated after this is merged and the code can then be changed to require/need both args (ie cleaned up).  If my assumption is wrong then change line 391?

Reply:
> `handle-pr-push` says `need` and not `optionally`

Whoops.

> I assume the gitgitgadget pipeline will be updated after this is merged and the code can then be changed to require/need both args (ie cleaned up).

Actually, I do not plan on updating the pipeline with ID 3 (i.e. the one servicing gitgitgadget/git).

Will change line 391, as you have suggested.

Reply:
I was reviewing individual commits when I first commented.  Looking at the merged set now and seeing the optional owner name (and error message wording issue) happens many times.  Would a commander option of --owner make this cleaner than an optional positional parameter?

btw, just noticed the use of `commander.gitGitWorkDir`.  Seems to ignore the `--work-dir` option (ie commander.workDir).

Reply:
> I was reviewing individual commits when I first commented. Looking at the merged set now and seeing the optional owner name (and error message wording issue) happens many times. Would a commander option of --owner make this cleaner than an optional positional parameter?

Having thought about your earlier comment about changing the pipeline to specify even `gitgitgadget`, I have come around to agreeing with your proposal.

As a consequence, I would like to keep this as a positional parameter, with the plan to make it mandatory after this PR was merged.

> btw, just noticed the use of `commander.gitGitWorkDir`. Seems to ignore the `--work-dir` option (ie commander.workDir).

You mean something like this?

```diff
diff --git a/script/misc-helper.ts b/script/misc-helper.ts
index 0446a9a..7af6591 100644
--- a/script/misc-helper.ts
+++ b/script/misc-helper.ts
@@ -24,7 +24,8 @@ if (commander.args.length === 0) {
 
 async function getGitGitWorkDir(): Promise<string> {
     if (!commander.gitGitWorkDir) {
-        commander.gitGitWorkDir = await gitConfig("gitgitgadget.workDir");
+        commander.gitGitWorkDir =
+            await gitConfig("gitgitgadget.workDir", commander.workDir || ".");
         if (!commander.gitGitWorkDir) {
             throw new Error(`Could not determine gitgitgadget.workDir`);
         }
```

I think you're right. I can sneak that in.